clang/lib/Driver/Driver.cpp

```diff
@@ -1430,6 +1430,18 @@ Compilation *Driver::BuildCompilation(ArrayRef<const char *> ArgList) {
   const ToolChain &TC = getToolChain(
       *UArgs, computeTargetTriple(*this, TargetTriple, *UArgs));
 
+  if (TC.getTriple().isAndroid()) {
+    llvm::Triple Triple = TC.getTriple();
+    unsigned TripleVersion = Triple.getEnvironmentVersion().getMajor();
+    StringRef TripleVersionName = Triple.getEnvironmentVersionString();
+
+    if (TripleVersion == 0 && TripleVersionName != "") {
```

Comment:
Check for `Triple.getEnvironmentVersion().empty()` instead of `.getMajor() == 0`.  In non-Android contexts, a major version of `0` may be valid.